constructor/winexe.py

```diff
@@ -259,6 +259,8 @@ def make_nsi(info, dir_path, extra_files=()):
     ppd['with_conclusion_text'] = bool(conclusion_text)
     ppd["enable_debugging"] = bool(os.environ.get("NSIS_USING_LOG_BUILD"))
     ppd["has_conda"] = info["_has_conda"]
+    ppd["custom_welcome"] = "welcome_file" in info and info["welcome_file"].endswith(".nsi")
+    ppd["custom_conclusion"] = "conclusion_file" in info and info["conclusion_file"].endswith(".nsi")
```

Comment:
This is a bit more succinct if you like it better:

```suggestion
    ppd["custom_welcome"] = info.get("welcome_file", "").endswith(".nsi")
    ppd["custom_conclusion"] = info.get("conclusion_file", "").endswith(".nsi")
```